### PR TITLE
Update docs to differentiate DeviceEvents and WindowEvents

### PR DESCRIPTION
**Pull request by alasher**
New to this project, thought I'd try to address #855. All input is appreciated! Many of the below regulations do not apply for this docfix, but I did generate the documentation with `cargo doc` and verified the links work appropriately. I changed the `WindowEvent` and `DeviceEvent` links to their dedicated docs page, instead of the sublink on the generic `Event` page. This way when the user clicks they can see all the different types of Window/Device, which I think makes more sense in the context they are linked in.

- [x] Tested on all platforms changed
- [x] `cargo fmt` has been run on this branch
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by Osspial** [comment]
Hi, and congratulations on the first PR!

There's one clarification I'd like to make the docs to make, but otherwise these seem like great changes. Right now, the patched documentation for `DeviceEvent` implies that mouse and keyboard events will *only* be dispatched through `DeviceEvent`. Instead, it'd be more accurate to say that *unfiltered* mouse and keyboard events get dispatched through `DeviceEvent`, as users will receive `WindowEvent`s for mouse and keyboard input (as filtered by the window manager) if the window is focused.

**Comment by AlterionX** [comment]
I had commented on the relevant issue, but an example of `WindowEvent` being used for keyboard input is here: https://docs.rs/winit/0.20.0-alpha1/winit/event/enum.WindowEvent.html#variant.KeyboardInput

Just for reference.

**Comment by alasher** [comment]
Thanks for the feedback! I've updated the description again. Hopefully it's not too wordy, but I think the text now accurately describes the key differences between DeviceEvents and WindowEvents. Feel free to nit-pick anything I wrote to make it more clear.

**Comment by Osspial** [comment]
That looks good. Thank you for looking into this!
